### PR TITLE
#2245 Better handle of mixed equation and added support for relatins

### LINKED ISSUE #2245: [Model Compilation] Improve handle of dynamic Y and F and enable relations from mixed types equations

### PR DESCRIPTION
**Pull request by rosiereflo**
within mixed equations
closes #2245

### CODE REVIEW

File: dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py [hunk 1]
```diff
@@ -424,62 +424,184 @@ def read_info_json(self):
...
+                    idx+=1
+                self.analyse_eq_expr(sub_eq[:-1], root)
+            elif expr.startswith("if") or expr.startswith("-if"):
+                idx = self.remove_condition(idx, splitted_eq)
+                node = self.Node()
+                node.eq = "IF"
+                root.children.append(node)
+                (final_expr, idx) = self.find_value(idx, splitted_eq)
+                self.analyse_eq_expr(final_expr, node)
+            elif expr.startswith("else"):
```
Comment (adriguir) [inline dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py:518]: No other possibilities there too to support, such as "- else" or anything similar?

**Review by adriguir — CHANGES_REQUESTED** [review]

File: dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py [hunk 2]
```diff
@@ -2959,17 +2959,12 @@ def prepare_for_evaldynamicytype(self):
...
-                            line = "   if (yType[ %s ] != DIFFERENTIAL) yType[ %s ] = %s;   /* %s (%s) */\n" % (str(ind), str(ind), "ALGEBRAIC", to_compile_name(v.get_name()), v.get_type())
-                            lines_to_insert.append(line)
-                        else:
-                            spin = "DIFFERENTIAL"
-                            line = "   yType[ %s ] = %s;   /* %s (%s) */\n" % (str(ind), spin, to_compile_name(v.get_name()), v.get_type())
-                            lines_to_insert.append(line)
-                    body = replace_equations_in_a_if_statement(eq.get_body_for_setf(), lines_to_insert, 4)
+                    line_to_insert_algebraic = "   if (yType[ %s ] != DIFFERENTIAL) yType[ %s ] = %s;   /* %s (%s) */\n" % (str(ind), str(ind), "ALGEBRAIC", to_compile_name(v.get_name()), v.get_type())
+                    spin = "DIFFERENTIAL"
+                    line_to_insert_differential = "   yType[ %s ] = %s;   /* %s (%s) */\n" % (str(ind), spin, to_compile_name(v.get_name()), v.get_type())
```
[the remark this answers is not in the document]
Comment (rosiereflo) [reply dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py]: gone in the latest version

File: dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py [hunk 3]
```diff
@@ -424,62 +424,184 @@ def read_info_json(self):
...
+                idx+=1
+                while nb_parenthesis > 0:
+                    if "(" in splitted_eq[idx]:
+                        nb_parenthesis+=splitted_eq[idx].count("(")
+                    if ")" in splitted_eq[idx]:
+                        nb_parenthesis-=splitted_eq[idx].count(")")
+                    sub_eq+= " " + splitted_eq[idx]
+                    idx+=1
+                self.analyse_eq_expr(sub_eq[:-1], root)
+            elif expr.startswith("if") or expr.startswith("-if"):
```
[the remark this answers is not in the document]
Comment (rosiereflo) [reply dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py:519]: you are right, never too cautious

File: dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py (hunk 1 above)
Comment (rosiereflo) [reply dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py:526]: no, the syntax is var1 -(if cond then var2 else -var3) so it is not possible to have a minus in front of a else

**Review by adriguir — APPROVED** [review]

**Pull request merged**